### PR TITLE
fix: fix build issue `‘maybe_unused’ attribute ignored` on centos server 

### PR DESCRIPTION
**Pull request by LittleFall**
Issue Number: close https://github.com/pingcap/tiflash/issues/4701

```release-note
None
```

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

File: dbms/src/Common/CPUAffinityManager.h [hunk 1]
```diff
@@ -115,8 +115,17 @@ class CPUAffinityManager
 #endif
 
     // unused except Linux
+
+#ifdef __GNUC__
+#  pragma GCC diagnostic push
+#  pragma GCC diagnostic ignored "-Wattributes"
+#endif
     [[maybe_unused]] int query_cpu_percent;
```
[the remark this answers is not in the document]
Comment (LittleFall) [reply dbms/src/Common/CPUAffinityManager.h]: addressed

File: libs/libcommon/include/common/defines.h [hunk 2]
```diff
@@ -199,3 +199,9 @@ static ALWAYS_INLINE inline void TIFLASH_NO_OPTIMIZE(T && var)
 #define TIFLASH_DUMMY_FUNCTION_DEFINITION
 #define tiflash_compiler_builtin_memcpy __builtin_memcpy
 #endif
+
+#ifdef __GNUC__
+#define UNUSED_MEMBER
+#else
```
Comment (fuzhe1989) [inline libs/libcommon/include/common/defines.h:205]: I'm not sure if it's better explicit define this macro for `__clang__`.

File: dbms/src/Common/CPUAffinityManager.h [hunk 3]
```diff
@@ -115,8 +117,9 @@ class CPUAffinityManager
 #endif
 
     // unused except Linux
-    [[maybe_unused]] int query_cpu_percent;
-    [[maybe_unused]] int cpu_cores;
+    UNUSED_MEMBER int query_cpu_percent;
```
Comment (solotzg) [inline dbms/src/Common/CPUAffinityManager.h:120]: Is it really unused? We'd better add symbol like `maybe`.

Comment (LittleFall) [reply dbms/src/Common/CPUAffinityManager.h:120]: addressed

File: libs/libcommon/include/common/defines.h (hunk 2 above)
Comment (LittleFall) [reply libs/libcommon/include/common/defines.h:205]: refined and tested in my mac and server.

**Review by fuzhe1989 — APPROVED** [review]

**Pull request merged**